### PR TITLE
Restrict All Docket Switch Attorney Task Actions to COTB Users

### PR DESCRIPTION
**Pull request by jcq**
Connects #15621

### Description
This updates `DocketSwitchMailTask`, `DocketSwitchDeniedTask`, and `DocketSwitchGrantedTask` to ensure that default task actions are also restricted to `ClerkOfTheBoard` users

### Acceptance Criteria
- [ ] Code compiles correctly

### CODE REVIEW

File: spec/models/tasks/docket_switch_mail_task_spec.rb [hunk 1]
```diff
@@ -31,16 +31,16 @@
 
       context "without docket_switch feature toggle" do
         it "returns the available_actions as defined by Task" do
-          expect(subject).to eq(task_actions)
+          expect(subject).to eq([])
```
Comment (leikkisa) [inline spec/models/tasks/docket_switch_mail_task_spec.rb:34]: I guess this it statement could be renamed

**Review by leikkisa — APPROVED** [review]
Hey JC - It might be good to differentiate OCB attorney tasks from regular attorney tasks through a parent class. There's only these two for now so that could also wait until we know if future OCB tasks will be similar.

There's one AC who's wording is a little confusing to me, but I'm picking up from the background and code that it's existing behavior.

File: spec/models/tasks/docket_switch_mail_task_spec.rb (hunk 1 above)
Comment (nanotone) [reply spec/models/tasks/docket_switch_mail_task_spec.rb:34]: `be_empty` to the rescue!

**Comment by nanotone** [comment]
Yeah, my inclination would be to insert a `DocketSwitchAttorneyTask` superclass, since the two files are 99% the same. The subclasses would just need to implement 2 one-liners.

**Comment by nanotone** [comment]
Two style nits:

- CC wants `default_actions` to be a class function. It's a little naggy now, but I think a class function would make a lot of sense if you went the superclass route, as it emphasizes the classy-ness of all this logic.
- In all 3 diffs, the `actions` variable sees a lot of mutation, through multiple layers of `if`s, which makes the code less readable. In these cases I'm a fan of "fail-fast" short-circuiting, i.e. `return [] unless ClerkOfTheBoard...`

**Comment by jcq** [comment]
> Yeah, my inclination would be to insert a `DocketSwitchAttorneyTask` superclass, since the two files are 99% the same. The subclasses would just need to implement 2 one-liners.

Oof, this seemed like a simple, elegant solution. It still should be, but turns out that if one tries to not implement `self.label` method in a subclass of `Task`, things break quickly (sorting in `TaskSorter` appears to happen on ALL tasks, not just relevant ones — and is dependent upon the label)

**Comment by leikkisa** [comment]
@jcq If you think it's better to not abstract out the super class, I can approve of the Code Climate warning that was complaining about duplicate code. I didn't realize this was a thing with tasks.

**Comment by jcq** [comment]
> @jcq If you think it's better to not abstract out the super class, I can approve of the Code Climate warning that was complaining about duplicate code. I didn't realize this was a thing with tasks.

I think it's OK — things are working now w/ the stub label that I gave it. However, it did surface what _feels_ like a bug in `TaskSorter` 🤔

**Pull request merged**